### PR TITLE
Fix MismatchedInputException in Ollama streaming

### PR DESCRIPTION
**Pull request by jpohlmeyer**
Handle MismatchedInputException in catch-clause.
It is either wrapped in ClientWebApplicationException with Status 400 or wrapped in ProcessingException depending on the Quarkus version.

See #685.

The diff display is a bit misleading, i only changed the catch and if clause + indentation.

Handling both cases in a single catch makes the if-clause a bit unwieldy, but two catch-clauses and extraction of the body to avoid code duplication was not really practical either.

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Thanks!

File: model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaRestApi.java [hunk 1]
```diff
@@ -85,40 +86,41 @@ class OllamaRestApiReaderInterceptor implements ReaderInterceptor {
...
-                                final var ctx = Vertx.currentContext();
-                                if (ctx == null) {
-                                    throw e;
-                                }
+            } catch (ClientWebApplicationException | ProcessingException e) {
+                if ((e instanceof ProcessingException pe && pe.getCause() instanceof MismatchedInputException) ||
+                        (e instanceof WebApplicationException wae
+                                && ((wae.getCause() instanceof JsonParseException && wae.getResponse().getStatus() == 200) ||
+                                        (wae.getCause() instanceof MismatchedInputException
+                                                && wae.getResponse().getStatus() == 400)))) {
```
Comment (geoand) [inline model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaRestApi.java:94]: Actually we should probably have a comment on why this is needed (i.e. the change in Quarkus).

Comment (jpohlmeyer) [reply model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaRestApi.java:94]: Yes, i will add a comment

Comment (geoand) [reply model-providers/ollama/runtime/src/main/java/io/quarkiverse/langchain4j/ollama/OllamaRestApi.java:96]: 🙏🏽

**Review by geoand — APPROVED** [review]

**Pull request merged**